### PR TITLE
Some Hybrid iSAM cleanup

### PR DESCRIPTION
**Pull request by dellaert**
- some naming conventions and spelling
- Made a more robust interface to Switching
- changed HybridGaussianISAM tests to reuse update graphs
- kept HybridNonlinearISAM tests the same but modernized them a bit
- in both cases, kept some regression but made tests have real teeth
- Added chain factors in testHybridEstimation

PS, ISAM test in latter produces BayesTrees like this, which seems right:

image

### CODE REVIEW

File: gtsam/hybrid/tests/Switching.h [hunk 1]
```diff
@@ -136,7 +145,7 @@ struct Switching {
    */
   Switching(size_t K, double between_sigma = 1.0, double prior_sigma = 0.1,
             std::vector<double> measurements = {},
-            std::string discrete_transition_prob = "1/2 3/2")
+            std::string transitionProbabilityTable = "1/2 3/2")
```
Comment (varunagrawal) [inline gtsam/hybrid/tests/Switching.h:148]: I thought variables were snake case and functions were camel case.

File: gtsam/hybrid/tests/testHybridGaussianISAM.cpp [hunk 2]
```diff
@@ -227,7 +180,7 @@ TEST(HybridGaussianElimination, Approx_inference) {
   }
 
   // Now we calculate the actual factors using full elimination
-  const auto [unprunedHybridBayesTree, unprunedRemainingGraph] =
+  const auto [unPrunedHybridBayesTree, unPrunedRemainingGraph] =
```
Comment (varunagrawal) [inline gtsam/hybrid/tests/testHybridGaussianISAM.cpp:183]: Nitpick: Isn't unpruned one word?

Source: https://www.oed.com/dictionary/unpruned_adj1?tl=true

**Review by varunagrawal — APPROVED** [review]
2 comments but LGTM

File: gtsam/hybrid/tests/Switching.h (hunk 1 above)
Comment (dellaert) [reply gtsam/hybrid/tests/Switching.h:148]: in google style, yes, but historically not in GTSAM.

File: gtsam/hybrid/tests/testHybridGaussianISAM.cpp (hunk 2 above)
Comment (dellaert) [reply gtsam/hybrid/tests/testHybridGaussianISAM.cpp:183]: eh. My code spellchecker complained. Will leave as is as otherwise another CI :-/

Comment (varunagrawal) [reply gtsam/hybrid/tests/testHybridGaussianISAM.cpp:183]: I can update this in a follow up. :)

File: gtsam/hybrid/HybridNonlinearISAM.cpp [hunk 3]
```diff
@@ -39,8 +39,8 @@ void HybridNonlinearISAM::update(const HybridNonlinearFactorGraph& newFactors,
   if (newFactors.size() > 0) {
     // Reorder and relinearize every reorderInterval updates
     if (reorderInterval_ > 0 && ++reorderCounter_ >= reorderInterval_) {
-      // TODO(Varun) Relinearization doesn't take into account pruning
-      reorder_relinearize();
+      // TODO(Varun) Re-linearization doesn't take into account pruning
+      reorderRelinearize();
```
Comment (varunagrawal) [inline gtsam/hybrid/HybridNonlinearISAM.cpp:43]: PS this name came from `NonlinearISAM` (the continuous only version).

Comment (varunagrawal) [reply gtsam/hybrid/HybridNonlinearISAM.cpp:43]: It's `NonlinearISAM::reorder_relinearize` there which is why we kept the same name.

**Pull request merged**